core/app/components/forms/elements/RelatedPubsElement.tsx

```diff
@@ -0,0 +1,203 @@
+"use client";
+
+import type { ReactNode } from "react";
+
+import { useMemo, useState } from "react";
+import { Value } from "@sinclair/typebox/value";
+import { useFieldArray, useFormContext } from "react-hook-form";
+import { relationBlockConfigSchema } from "schemas";
+
+import type { JsonValue } from "contracts";
+import type { InputComponent, PubsId } from "db/public";
+import { Button } from "ui/button";
+import { FormControl, FormDescription, FormField, FormItem, FormLabel, FormMessage } from "ui/form";
+import { Plus, Trash } from "ui/icon";
+import { MultiBlock } from "ui/multiblock";
+import { Popover, PopoverContent, PopoverTrigger } from "ui/popover";
+
+import type { PubFieldFormElementProps } from "../PubFieldFormElement";
+import type { ElementProps } from "../types";
+import type { GetPubsResult } from "~/lib/server";
+import { AddRelatedPubsPanel } from "~/app/components/forms/AddRelatedPubsPanel";
+import { useContextEditorContext } from "../../ContextEditor/ContextEditorContext";
+import { useFormElementToggleContext } from "../FormElementToggleContext";
+import { PubFieldFormElement } from "../PubFieldFormElement";
+
+const RelatedPubBlock = ({
+	pub,
+	onRemove,
+	valueComponentProps,
+	slug,
+}: {
+	pub: Pick<GetPubsResult[number], "title" | "id">;
+	onRemove: () => void;
+	valueComponentProps: PubFieldFormElementProps;
+	slug: string;
+}) => {
+	const { title, id } = pub;
+
+	return (
+		<div className="flex items-center justify-between rounded border border-l-[12px] border-l-emerald-100 p-3">
+			<div className="flex flex-col items-start gap-1 text-sm">
+				<span className="font-semibold">{title || id}</span>
+				<ConfigureRelatedValue {...valueComponentProps} slug={slug} />
```

Comment:
I think we should use the pubtitle here as well
```suggestion
				<span className="font-semibold">{getPubTitle(pub)}</span>
```